### PR TITLE
Organize notebooks on the RTD examples page

### PR DESCRIPTION
**Pull request by rocco8773**
We finally have enough example Jupyter notebooks that the example's page on RTD deserves some organization.  This PR does...

1. Organize the layout of the notebooks on the RTD examples page.
2. Organize and structure the `/docs/notebooks/` directory to make the RTD examples page more maintainable (i.e. less work for us).
3. Repair and/or add some documentation links to the notebooks.
4. Update out of date `intersphinx` mappings.
5. Update the `nbsphinx_thumbnails` config variable in `conf.py` so the nested notebooks in `/docs/notebooks/*/*` still have a fallback thumbnail image.

This is just a first take at organizing the notebooks.  I'm sure we'll end up making more structural changes as notebooks are added, but I think this is still a good organization to have implemented before the `v0.6` release.

### CODE REVIEW

**Comment by rocco8773** [comment]
@StanczakDominik I set this up to auto-merge, so whenever you click approve it'll do its thing.

**Comment by StanczakDominik** [comment]
And good morning to you too! ;)

Looks like you're still tinkering with it a little? :)

**Comment by rocco8773** [comment]
Good morning! :)

I see you just merged PR #1065, so I'll need to make the patch and then this will be ready to go.

**Review by StanczakDominik — APPROVED** [review]

File: docs/examples.rst [hunk 1]
```diff
@@ -2,20 +2,61 @@ Examples
...
-General examples
-----------------
+.. contents::
+   :local:
 
-General-purpose and introductory examples from PlasmaPy.
+General
+-------
 
-`You may also run these online via Binder (link uses the current master version). <https://mybinder.org/v2/gh/PlasmaPy/PlasmaPy/master?filepath=plasmapy/docs/notebooks>`_
```
[the remark this answers is not in the document]
Comment (rocco8773) [reply docs/examples.rst:12]: Yeah, the link was not working and each notebook has it's own link, so I thought it was unnecessary.  It's easily put back if deemed needed.

**Pull request merged**